pkg/scheduler/in_memory_build_queue.go

```diff
@@ -1857,14 +1859,25 @@ func (i *invocation) getInvocationState(bq *InMemoryBuildQueue) *buildqueuestate
 			activeInvocationsCount++
 		}
 	}
+
+	recursiveQueuedOperationsCount := uint32(0)
+	children := slices.Collect(maps.Values(i.children))
+	for len(children) > 0 {
+		child := children[0]
+		children = children[1:]
+		recursiveQueuedOperationsCount += uint32(child.queuedOperations.Len())
+		children = append(children, slices.Collect(maps.Values(child.children))...)
+	}
```

Comment:
Given that the tree of invocations is rather shallow, wouldn't a simple DFS implementation be better/faster/simpler?